src/main/java/io/lbry/browser/LbrynetService.java

```diff
@@ -118,6 +158,188 @@ protected void doStartForeground(Bundle extras) {
+                public void run() {
+                    LbrynetService.this.pollFileList();
+                }
+            }, 0, SDK_POLL_INTERVAL, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void pollFileList() {
+        BufferedReader reader = null;
+        DataOutputStream dos = null;
+        HttpURLConnection conn = null;
+
+        try {
+            JSONObject request = new JSONObject();
+            request.put("method", "file_list");
+
+            URL url = new URL(SDK_URL);
+            conn = (HttpURLConnection) url.openConnection();
+            conn.setDoInput(true);
+            conn.setDoOutput(true);
+            conn.setUseCaches(false);
+            conn.setRequestMethod("POST");
+            conn.setRequestProperty("Content-type", "application/json");
+
+            dos = new DataOutputStream(conn.getOutputStream());
+            dos.writeBytes(request.toString());
+            dos.flush();
+            dos.close();
+
+            if (conn.getResponseCode() == 200) {
+                reader = new BufferedReader(new InputStreamReader(conn.getInputStream(), "utf-8"));
+                StringBuilder sb = new StringBuilder();
+                String input;
+                while ((input = reader.readLine()) != null) {
+                    sb.append(input);
+                }
+
+                JSONObject response = new JSONObject(sb.toString());
+                if (!response.has("error")) {
+                    handlePollFileResponse(response);
+                }
+            } else {
+                reader = new BufferedReader(new InputStreamReader(conn.getErrorStream(), "utf-8"));
+                StringBuilder sb = new StringBuilder();
+                String error;
+                while ((error = reader.readLine()) != null) {
+                    sb.append(error);
+                }
+                android.util.Log.d(TAG, sb.toString());
+            }
+        } catch (IOException ex) {
+            android.util.Log.e(TAG, ex.getMessage(), ex);
+            // ignore and continue
+        } catch (Exception ex) {
+            android.util.Log.e(TAG, ex.getMessage(), ex);
+            // ignore
+        } finally {
+            try {
+                if (reader != null) {
+                    reader.close();
+                }
+                if (conn != null) {
+                    conn.disconnect();
+                }
+            } catch (IOException ex) {
+                // pass
+            }
+        }
+    }
+
+    private void handlePollFileResponse(JSONObject response) {
+        Context context = getApplicationContext();
+        if (response.has("result")) {
+            JSONArray fileItems = response.optJSONArray("result");
+            if (fileItems != null) {
+                try {
+                    List<String> itemUris = new ArrayList<String>();
+                    for (int i = 0; i < fileItems.length(); i++) {
+                        JSONObject item = fileItems.getJSONObject(i);
```

Comment:
Looks like you could use the reverse of https://github.com/havardh/javaflow

Reply:
I continue to think it would be really cool if there was a way for a single definition of `Claim`, `File`, etc. to be propagated from the SDK to a variety of languages. We ourselves would presumably use it in JavaScript, Java, Go. (@lyoshenka @eukreign no need to comment, but pointing out a use-case)